### ISSUE TITLE
non-required properties are never rendered

### ISSUE THREAD

**Issue by ducin** [json-schema-faker/json-schema-faker#179]
This [example schema](http://json-schema-faker.js.org/#gist/d51424c56a7923760ba6329167dcaf90) comes from [the official docs](http://json-schema.org/examples.html), so it'd be really cool to support it better than we do now. I expect to have non-required fields generated randomly (50:50 would be good I think). And non-required properties are never included in the result.

**Comment by ducin** [comment]
This one will be closed after the `v0.3.6` release.

**Comment by chinclubi** [comment]
As you mention above ... "non-required fields generated randomly"

It's quite hard to test a non-required field.

I suggest you to create a config in option which can enable or disable this randomize.

Ex. If this option is enabled, 50/50 is good. In the other hands, a non-required field will be generated 100%.

Thank you..

**Comment by pateketrueke** [comment]
> tl;dr - I rewrote how properties are merged/shuffled before faking the whole object, I consider this is solved now but certainly it could be perfect.

Enabling this feature is tightly related with #111 and shall be awesome!

I'm sure we can focus on this kind of issues since the library is stable now. (:
